tests/v2/integration/catalogv2/cluster_repo_test.go

```diff
@@ -799,6 +798,9 @@ func (c *ClusterRepoTestSuite) testClusterRepoRetries(params ClusterRepoParams)
 	require.NoError(c.T(), err)
 
 	downloadTime := cr.Status.DownloadTime
+
+	cr, err = c.catalogClient.ClusterRepos().Get(context.TODO(), cr.Name, metav1.GetOptions{})
+	require.NoError(c.T(), err)
```

Comment:
Could we backport this to `release/v2.9`?
```
    cluster_repo_test.go:822: 
        	Error Trace:	/home/runner/_work/rancher/rancher/tests/v2/integration/catalogv2/cluster_repo_test.go:822
        	            				/home/runner/_work/rancher/rancher/tests/v2/integration/catalogv2/cluster_repo_test.go:143
        	Error:      	Received unexpected error:
        	            	Operation cannot be fulfilled on clusterrepos.catalog.cattle.io "test-git-small-fork-cluster-repo": the object has been modified; please apply your changes to the latest version and try again
        	Test:       	TestClusterRepoTestSuite/TestGitRepoRetries
--- FAIL: TestClusterRepoTestSuite (26.02s)
    --- PASS: TestClusterRepoTestSuite/TestGitRepo (10.54s)
    --- FAIL: TestClusterRepoTestSuite/TestGitRepoRetries (12.22s)
```
https://github.com/rancher/rancher/actions/runs/11140137938/job/30959250592

Reply:
Also, consider using [`retry.RetryOnConflict`](https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict)